### PR TITLE
Swift optimizations

### PR DESCRIPTION
**Pull request by mbrandonw**
While trying to archive the app I noticed that there was a compiler crash when linking KsApi :( That sent me down a path to not only fix that, but also get some swift compiler optimizations done so that i could better test what was happening.

Some rough numbers I took:

Before optimizations: 1min 22secs (debug build), ~15 mins (archive build)
After optimizations: 45 secs (debug build), 2mins 30 secs (archive build)

Pretty significant! There are two things I did:

* Turned on xcode warnings for functions that take a long time to compile. I set the threshold to 500ms cause we have too many that take over 100ms, but that allowed me to find quite a few poor performers, including one that was taking 26 seconds!
* I noticed that `MockService` was taking 5+ mins to compile and another 5+ mins for code gen for release, and that was one of the reasons an archive took 15+ mins. The fix was simply to exclude it from building for release by doing `#if DEBUG`! Compile times sped up so much that I was even able to re-enable WMO, which @stephencelis and I couldn't figure out why it was taking so long many months ago.

### CODE REVIEW

File: KsApi/models/Backing.swift [hunk 1]
```diff
@@ -36,16 +36,17 @@ public func == (lhs: Backing, rhs: Backing) -> Bool {
 extension Backing: Decodable {
   public static func decode(_ json: JSON) -> Decoded<Backing> {
     let create = curry(Backing.init)
-    let tmp = create
+    let tmp1 = create
```
Comment (mbrandonw) [inline KsApi/models/Backing.swift]: argo was of course the cause of quite a few slow functions, so i did a lot of work on the slowest ones...

File: KsApi/models/DiscoveryParams.swift [hunk 2]
```diff
@@ -94,25 +94,27 @@ extension DiscoveryParams: CustomStringConvertible, CustomDebugStringConvertible
 extension DiscoveryParams: Decodable {
   public static func decode(_ json: JSON) -> Decoded<DiscoveryParams> {
     let create = curry(DiscoveryParams.init)
+
     let tmp1 = create
-      <^> (json <|? "backed" >>- stringIntToBool)
+      <^> ((json <|? "backed" >>- stringIntToBool) as Decoded<Bool?>)
```
Comment (mbrandonw) [inline KsApi/models/DiscoveryParams.swift]: this function took nearly 30 seconds to compile! there are a few ways we could approach this but i decided to just help out the compiler with some of the flat maps by telling it what typeof decoded value i expected

**Comment by theginbin** [comment]
super!

File: KsApi.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -2405,7 +2405,7 @@
 				IPHONEOS_DEPLOYMENT_TARGET = 9.0;
 				MTL_ENABLE_DEBUG_INFO = YES;
 				ONLY_ACTIVE_ARCH = YES;
-				OTHER_SWIFT_FLAGS = "-Xfrontend -debug-time-function-bodies";
+				OTHER_SWIFT_FLAGS = "-Xfrontend -warn-long-function-bodies=500 -D DEBUG";
```
Comment (stephencelis) [inline KsApi.xcodeproj/project.pbxproj]: 🎉

File: KsApi/models/Backing.swift (hunk 1 above)
Comment (stephencelis) [inline KsApi/models/Backing.swift]: 😢 Wish there were a better way.

**Pull request merged**